libs/agno/tests/integration/workflow/test_streaming_events.py

```diff
@@ -0,0 +1,851 @@
+# STEP EVENTS TESTS
+# ============================================================================
+
+def test_step_events_with_stream_intermediate_steps_false(workflow_storage):
+    """Test Step events with stream_intermediate_steps=False."""
+    workflow = Workflow(
+        name="Step Events Test",
+        storage=workflow_storage,
+        steps=[Step(name="test_step", executor=step_function)],
+    )
+
+    events = list(workflow.run(message="test", stream=True, stream_intermediate_steps=False))
+    event_types = [type(event).__name__ for event in events]
+
+    # Should only have workflow events, NO step started/completed events
+    assert "WorkflowStartedEvent" in event_types
+    assert "WorkflowCompletedEvent" in event_types
+    assert "StepStartedEvent" not in event_types
+    assert "StepCompletedEvent" not in event_types
+
+
+def test_step_events_with_stream_intermediate_steps_true(workflow_storage):
+    """Test Step events with stream_intermediate_steps=True."""
+    workflow = Workflow(
+        name="Step Events Test",
+        storage=workflow_storage,
+        steps=[Step(name="test_step", executor=step_function)],
+    )
+
+    events = list(workflow.run(message="test", stream=True, stream_intermediate_steps=True))
+
+    # Check for step events
+    step_started = [e for e in events if isinstance(e, StepStartedEvent)]
+    step_completed = [e for e in events if isinstance(e, StepCompletedEvent)]
+    workflow_started = [e for e in events if isinstance(e, WorkflowStartedEvent)]
+    workflow_completed = [e for e in events if isinstance(e, WorkflowCompletedEvent)]
+
+    assert len(step_started) == 1
+    assert len(step_completed) == 1
+    assert len(workflow_started) == 1
+    assert len(workflow_completed) == 1
+
+    # Verify event details
+    assert step_started[0].step_name == "test_step"
+    assert step_completed[0].step_name == "test_step"
+
+
+def test_workflow_events_with_stream_intermediate_steps_false(workflow_storage):
+    """Test that with stream_intermediate_steps=False, only workflow and content events are received."""
+    workflow = Workflow(
+        name="Test Workflow",
+        steps=[
+            Step(name="step_1", executor=simple_step_1),
+            Step(name="step_2", executor=simple_step_2),
+        ],
+        storage=workflow_storage,
+    )
+
+    events = list(workflow.run("test message", stream=True, stream_intermediate_steps=False))
+
+    # Extract event types
+    event_types = [type(event).__name__ for event in events]
+
+    # Should only have workflow events and step outputs, NO step started/completed events
+    assert "WorkflowStartedEvent" in event_types
+    assert "WorkflowCompletedEvent" in event_types
+    assert "StepStartedEvent" not in event_types
+    assert "StepCompletedEvent" not in event_types
+
+    # Should have step outputs from the function executors
+    step_outputs = [event for event in events if isinstance(event, StepOutput)]
```

Comment:
We should reprocess the `StepOutput` as a `StepOutputEvent` (since it would need the other event stuff, like workflow_id, etc?